pages/api/download-subdirectory.ts

```diff
@@ -12,19 +12,33 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     const array = await response.arrayBuffer();
 
     const zip = await JSZip.loadAsync(array, {});
-    const rootName = Object.keys(zip.files)[0];
-    const subDirZip = zip.folder(rootName)?.folder(data.subdirectory);
 
-    if (subDirZip === null || subDirZip === undefined) {
+    const notFound = Object.keys(zip.files).every((fileName) => {
+      const firstLevelDirectory = fileName.split('/')[1];
+      return firstLevelDirectory !== data.subdirectory;
+    });
+    if (notFound) {
       throw TypeError('subdirectory does not exist');
     }
-    const base64send = await subDirZip.generateAsync({ type: 'base64' });
+
+    const rootName = Object.keys(zip.files)[0];
+    const subdirectoryZip = zip.folder(rootName)?.folder(data.subdirectory);
+
+    if (subdirectoryZip == null ) {
```

Comment:
```suggestion
    if (subdirectoryZip == null) {
```

for prettier linting to pass